### PR TITLE
feat: Add FluentNumberInput<TValue> component with demo and tests

### LINKED ISSUE #4667: [Feature Request] Add FluentNumberInput component for V5
## Summary

The V5 release of Fluent UI Blazor does not include a dedicated `FluentNumberInput` component for numeric input. In V4, numeric input was handled via `FluentNumberField`, but there is no equivalent in V5.

A `FluentNumberInput<TValue>` component wrapping `<fluent-text-input type="number">` with proper step/min/max support would be very useful.

## Use Case

Entering numeric values (integers, decimals) with validation, step increment, and min/max boundaries is a common requirement in forms.

## Expected API / Code Example

```razor
@using Microsoft.FluentUI.AspNetCore.Components

<FluentNumberInput TValue="int"
                   @bind-Value="@Quantity"
                   Min="0"
                   Max="100"
                   Step="1"
                   Placeholder="Enter quantity"
                   Label="Quantity"
                   Required="true" />

<FluentNumberInput TValue="decimal"
                   @bind-Value="@Price"
                   Min="0.00m"
                   Max="9999.99m"
                   Step="0.01m"
                   Placeholder="Enter price"
                   Label="Price"
                   Width="200px" />

@code {
    private int Quantity { get; set; } = 1;
    private decimal Price { get; set; } = 9.99m;
}
```

### Expected Parameters

| Parameter | Type | Description |
|-----------|------|-------------|
| `TValue` | Generic (`int`, `double`, `decimal`, etc.) | The numeric type for the value |
| `Value` | `TValue` | The bound value |
| `Min` | `TValue?` | Minimum allowed value |
| `Max` | `TValue?` | Maximum allowed value |
| `Step` | `TValue?` | Step increment (e.g., `0.01` for decimals) |
| `Placeholder` | `string?` | Hint text shown when empty |
| `Appearance` | `TextInputAppearance` | Visual style (Outline, FilledLighter, etc.) |
| `StartTemplate` | `RenderFragment?` | Content prefix (e.g., currency symbol) |
| `EndTemplate` | `RenderFragment?` | Content suffix (e.g., unit label) |

## Additional Context

This component should support:
- All common numeric types (`byte`, `short`, `int`, `long`, `float`, `double`, `decimal`)
- Proper invariant culture parsing (HTML `<input type="number">` uses `.` as decimal separator regardless of locale)
- Integration with `FluentField` for label, validation messages, and required state
- Immediate mode (update on each keystroke) and standard mode (update on blur/change)

### PR DESCRIPTION
**Pull request by AClerbois**
## Summary

Adds a new `FluentNumberInput<TValue>` component with full JS interop support, demo documentation, and unit tests.

### Changes

**Component** (`src/Core/Components/NumberInput/`)
- `FluentNumberInput<TValue>` — numeric input wrapping `<fluent-text-input type="number">` inside `FluentField`
- Supports all .NET numeric types: `byte`, `sbyte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `float`, `double`, `decimal`
- Invariant-culture formatting/parsing (HTML spec compliance)
- JS interop to apply `step`/`min`/`max` to the inner shadow-DOM `<input>`, with re-application on parameter changes

**JS Interop** (`src/Core.Scripts/`)
- `NumberInput.ts` — helper to set `step/min/max` on the inner control
- Registered in `ExportedMethods.ts`

**Demo documentation** (`examples/Demo/.../NumberInput/`)
- Documentation page with examples: Numeric Types, Culture, Appearance, States, Prefix/Suffix
- API section auto-generated via `{{ API Type=FluentNumberInput<int> }}`

**Tests** (`tests/Core/Components/NumberInput/`)
- Unit tests covering rendering, binding, parsing, culture-aware parsing, disabled/readonly states, templates, and label
- Uses both Verify snapshot and MarkupMatches patterns

Closes #4667

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ tests/
 └── Integration/                    # Integration tests
 
 examples/
-├── Demo/                           # Demo application
```
Comment (dvoituron) [inline AGENTS.md:28]: Why this change?

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputAppearance.razor [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿<FluentStack HorizontalGap="20px">
```
Comment (dvoituron) [inline examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputAppearance.razor]: Add the `Wrap` parameter to set the UI responsive

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                       @bind-Value="@price" />
+</FluentStack>
+
+<FluentLabel>
+    Value = @price
+    — Formatted (CurrentCulture): @price.ToString("N2")
+    — Formatted (InvariantCulture): @price.ToString("N2", CultureInfo.InvariantCulture)
+</FluentLabel>
+
+<blockquote style="margin-top: 1rem;">
```
Comment (dvoituron) [inline examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor]: The doc must be in a md file

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                       Min="0"
+                       Max="9999.99"
+                       Placeholder="0.00"
+                       @bind-Value="@price" />
+</FluentStack>
+
+<FluentLabel>
+    Value = @price
+    — Formatted (CurrentCulture): @price.ToString("N2")
+    — Formatted (InvariantCulture): @price.ToString("N2", CultureInfo.InvariantCulture)
```
Comment (dvoituron) [inline examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor]: I don't understand why these 2 formats. C# number don't have the distinction between local/invariant (only strings)

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputDefault.razor [hunk 5]
```diff
@@ -0,0 +1,8 @@
+﻿<FluentNumberInput TValue="int" Label="Quantity" @bind-Value="@quantity" />
```
Comment (dvoituron) [inline examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputDefault.razor]: Why this example? No interest: you already explained how to bind a number.

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputImmediate.razor [hunk 6]
```diff
@@ -0,0 +1,16 @@
+﻿<FluentNumberInput TValue="double"
```
Comment (dvoituron) [inline examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputImmediate.razor]: Same: not interesting since a similar example already exists

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputMinMaxStep.razor [hunk 7]
```diff
@@ -0,0 +1,34 @@
+﻿<FluentStack Orientation="Orientation.Vertical" VerticalGap="10">
```
Comment (dvoituron) [inline examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputMinMaxStep.razor]: Same: to delete

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: tests/Core/Components/NumberInput/FluentNumberInputTests.razor [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+    [Fact]
+    public void FluentNumberInput_Default_Int()
+    {
+        // Arrange && Act
+        using var context = new IdentifierContext(i => "myId");
+        var cut = Render(@<FluentNumberInput TValue="int" Value="5" />);
+
+        // Assert
+        cut.Find("fluent-text-input")
+           .MarkupMatches("<fluent-text-input id=\"myId\" slot=\"input\" appearance=\"outline\" inputmode=\"decimal\" type=\"number\" value=\"5\" />");
```
Comment (dvoituron) [inline tests/Core/Components/NumberInput/FluentNumberInputTests.razor]: Can you check the HTML using the Verify method?

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/FluentNumberInput.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+## Binding with ImmediateDelay
+
+You can bind the value and update immediately as the user types by setting
+the `Immediate` and `ImmediateDelay` properties.
+
+{{ NumberInputImmediate }}
+
+## API FluentNumberInput
+
+{{ API Type=FluentNumberInput<int> }}
```
[the remark this answers is not in the document]
Comment (AClerbois) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/FluentNumberInput.md:73]: Fixed

File: src/Core/Components/NumberInput/FluentNumberInput.razor.cs [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+            await InitializeImmediateAsync();
+
+            // Set step/min/max on the inner <input> inside the shadow DOM
+            await JSRuntime.InvokeVoidAsync(
+                "Microsoft.FluentUI.Blazor.Components.NumberInput.setNumberAttributes",
+                Id,
+                FormatNullableValue(Step),
+                FormatNullableValue(Min),
+                FormatNullableValue(Max));
+        }
```
[the remark this answers is not in the document]
Comment (AClerbois) [reply src/Core/Components/NumberInput/FluentNumberInput.razor.cs:164]: Fixed

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputAppearance.razor (hunk 2 above)
Comment (AClerbois) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputAppearance.razor]: Done

File: AGENTS.md (hunk 1 above)
Comment (AClerbois) [reply AGENTS.md:28]: It's was a mess with the older

File: AGENTS.md [hunk 11]
```diff
@@ -25,7 +25,7 @@ tests/
...
+├── Demo/                           # Demo application and provide a self-documentation
```
[the remark this answers is not in the document]
Comment (AClerbois) [reply AGENTS.md:28]: Fixed

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor (hunk 3 above)
Comment (AClerbois) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor]: Fixed

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputDefault.razor (hunk 5 above)
Comment (AClerbois) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputDefault.razor]: You're right

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputMinMaxStep.razor (hunk 7 above)
Comment (AClerbois) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputMinMaxStep.razor]: Fixed

File: tests/Core/Components/NumberInput/FluentNumberInputTests.razor (hunk 8 above)
Comment (AClerbois) [reply tests/Core/Components/NumberInput/FluentNumberInputTests.razor]: DOne

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputImmediate.razor (hunk 6 above)
Comment (AClerbois) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputImmediate.razor]: Fixed

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor (hunk 4 above)
Comment (AClerbois) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor]: This help besoin to see the diff between the browser settings and the application one

File: src/Core/Components/NumberInput/FluentNumberInput.razor.cs [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+
+        if (BindConverter.TryConvertTo<TValue>(normalized, CultureInfo.InvariantCulture, out var parsedValue))
+        {
+            result = parsedValue;
+            validationErrorMessage = null;
+            return true;
+        }
+
+        result = default;
+        validationErrorMessage = $"The '{DisplayName ?? "Unknown Bound Field"}' field is not valid.";
```
Comment (MarvinKlein1508) [inline src/Core/Components/NumberInput/FluentNumberInput.razor.cs]: this needs to be in the localization files

File: src/Core/Components/NumberInput/FluentNumberInput.razor.cs [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+                currentMax);
+        }
+    }
+
+    /// <summary>
+    /// Formats the value as a string using InvariantCulture.
+    /// </summary>
+    protected override string? FormatValueAsString(TValue value)
+    {
+        return Convert.ToString(value, CultureInfo.InvariantCulture);
```
Comment (MarvinKlein1508) [inline src/Core/Components/NumberInput/FluentNumberInput.razor.cs]: we should use BindConverter here to avoid boxing.

```cs
protected override string? FormatValueAsString(TValue? value)
    {
        // Avoiding a cast to IFormattable to avoid boxing.
        switch (value)
        {
            case null:
                return null;

            case int @int:
                return BindConverter.FormatValue(@int, CultureInfo.InvariantCulture);

            case long @long:
                return BindConverter.FormatValue(@long, CultureInfo.InvariantCulture);

            case short @short:
                return BindConverter.FormatValue(@short, CultureInfo.InvariantCulture);

            case float @float:
                return BindConverter.FormatValue(@float, CultureInfo.InvariantCulture);

            case double @double:
                return BindConverter.FormatValue(@double, CultureInfo.InvariantCulture);

            case decimal @decimal:
                return BindConverter.FormatValue(@decimal, CultureInfo.InvariantCulture);

            default:
                throw new InvalidOperationException($"Unsupported type {value.GetType()}");
        }
```

File: src/Core/Components/NumberInput/FluentNumberInput.razor.cs [hunk 14]
```diff
@@ -0,0 +1,226 @@
...
+    /// <see cref="CultureInfo.InvariantCulture"/> to avoid misinterpretation
+    /// (e.g. in fr-FR, <c>.</c> is the thousands separator, so <c>"2.5"</c> would be parsed as <c>25</c>).
+    /// </summary>
+    [SuppressMessage("Trimming", "IL2091:Target generic argument does not satisfy 'DynamicallyAccessedMembersAttribute' in target method or type. The generic parameter of the source method or type does not have matching annotations.",
+                 Justification = "In the context, the 'TValue' will not be trimmed.")]
+    protected override bool TryParseValueFromString(string? value, [MaybeNullWhen(false)] out TValue result, [NotNullWhen(false)] out string? validationErrorMessage)
+    {
+        // Normalize comma to dot to handle edge cases where a browser
+        // might send the locale-formatted decimal separator.
+        var normalized = value?.Replace(',', '.');
```
Comment (MarvinKlein1508) [inline src/Core/Components/NumberInput/FluentNumberInput.razor.cs:186]: I don't think that this is necessary. The native `InputNumber` component from Blazor just does it like this:
```cs
protected override bool TryParseValueFromString(string? value, [MaybeNullWhen(false)] out TValue result, [NotNullWhen(false)] out string? validationErrorMessage)
    {
        if (BindConverter.TryConvertTo<TValue>(value, CultureInfo.InvariantCulture, out result))
        {
            validationErrorMessage = null;
            return true;
        }
        else
        {
            validationErrorMessage = string.Format(CultureInfo.InvariantCulture, ParsingErrorMessage, DisplayName ?? FieldIdentifier.FieldName);
            return false;
        }
    }
```

File: src/Core/Components/NumberInput/FluentNumberInput.razor.cs [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+public partial class FluentNumberInput<TValue> : FluentInputImmediateBase<TValue>, IFluentComponentElementBase, ITooltipComponent
+    where TValue : struct, IComparable<TValue>
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FluentNumberInput{TValue}"/> class.
+    /// </summary>
+    /// <exception cref="InvalidOperationException">Thrown when TValue is not a supported numeric type.</exception>
+    public FluentNumberInput(LibraryConfiguration configuration) : base(configuration)
+    {
+        if (typeof(TValue) != typeof(byte) &&
```
Comment (MarvinKlein1508) [inline src/Core/Components/NumberInput/FluentNumberInput.razor.cs]: we should probably check for nullable types as well.
```cs
 var targetType = Nullable.GetUnderlyingType(typeof(TValue)) ?? typeof(TValue);
```

File: tests/Core/Components/NumberInput/FluentNumberInputTests.razor [hunk 16]
```diff
@@ -0,0 +1,221 @@
+﻿@using Microsoft.FluentUI.AspNetCore.Components.Utilities
+@using System.Globalization
+@using Xunit;
+@inherits FluentUITestContext
+@code
+{
+    public FluentNumberInputTests()
```
Comment (MarvinKlein1508) [inline tests/Core/Components/NumberInput/FluentNumberInputTests.razor:7]: can you add test cases for `null` as well? In v4 this component was quite buggy with null.

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor [hunk 17]
```diff
@@ -0,0 +1,18 @@
+﻿@using System.Globalization
+
+<FluentStack Orientation="Orientation.Vertical" VerticalGap="10">
+    <FluentNumberInput TValue="double"
```
Comment (dvoituron) [inline examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor:4]: You must add a `Culture` (type CultureInfo) parameter to allow a specific format when the browser or server uses a different culture. E.g. https://localhost:7197/DateTime/Calendar#culture

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+                       Label="@($"Culture: {CultureInfo.CurrentCulture.Name}")"
+                       Step="0.01"
+                       Min="0"
+                       Max="9999.99"
```
Comment (dvoituron) [inline examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor:8]: If the minimum and maximum values are not met, the bound value must be set to the minimum or maximum value, not to the specified value.
See an example here: https://www.mudblazor.com/components/numericfield

Image

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+                       Placeholder="0.00"
+                       @bind-Value="@price" />
```
Comment (dvoituron) [inline examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor:10]: I don't know why, but decimal characters aren't accepted in this component, like in the `input` element by default.
Example: 
1. Using `.` or `,` in the FluentNumberInput
2. Using the same keys in `<input type="number" step="0.01" min="0" max="999" @bind-value="@price" />`

Image

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor (hunk 19 above)
Comment (MarvinKlein1508) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor:10]: @dvoituron I can't even write `,` or `.` into this fields. How did you put the `,` there?

Comment (dvoituron) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor:10]: Maybe because my Windows system is set to use a comma… and the “Group+Decimal” symbols are accepted in the “input” field?

{CBE6D7CD-A8FF-4017-9492-39B19CE39840}

Comment (MarvinKlein1508) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor:10]: @AClerbois it looks like the underlying web-component is not yet supporting the `number` type. The only way for us to get this to work is switching the input type to text.

File: tests/Core/Components/NumberInput/FluentNumberInputTests.razor (hunk 16 above)
Comment (AClerbois) [reply tests/Core/Components/NumberInput/FluentNumberInputTests.razor:7]: good remark, I will do

File: src/Core/Components/NumberInput/FluentNumberInput.razor.cs (hunk 15 above)
Comment (AClerbois) [reply src/Core/Components/NumberInput/FluentNumberInput.razor.cs]: Resolved by switching the constraint to `where TValue : struct, INumber<TValue>`. This provides compile-time safety — only numeric types are accepted. The 11-type runtime check and `Nullable.GetUnderlyingType` are no longer needed since the compiler enforces the constraint.

File: tests/Core/Components/NumberInput/FluentNumberInputTests.razor (hunk 16 above)
Comment (AClerbois) [reply tests/Core/Components/NumberInput/FluentNumberInputTests.razor:7]: Added tests for invalid input (`"abc"`) and empty input (`""`), which verify the value remains unchanged when parsing fails. With the `struct` constraint, `TValue` itself cannot be `null` (e.g. `int` defaults to `0`). Total: 24 tests, 95.5% coverage.

File: src/Core/Components/NumberInput/FluentNumberInput.razor.cs (hunk 14 above)
Comment (AClerbois) [reply src/Core/Components/NumberInput/FluentNumberInput.razor.cs:186]: Agreed. Replaced `BindConverter.TryConvertTo` with `TValue.TryParse(normalized, Culture, out var parsedValue)` — a static abstract method from `IParsable<TSelf>` (inherited by `INumber<TValue>`). Simpler and no trimming annotation needed.

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor (hunk 18 above)
Comment (AClerbois) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor:8]: Done. Values are now clamped to Min/Max bounds in `TryParseValueFromString`. Added tests: `Clamp_AboveMax`, `Clamp_BelowMin`, `Clamp_WithinRange`, `Clamp_Double_AboveMax`, and `Clamp_Double_BelowMin`.

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor (hunk 17 above)
Comment (AClerbois) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor:4]: Done. Added `[Parameter] public CultureInfo Culture { get; set; } = CultureInfo.InvariantCulture;` — used in `FormatValueAsString`, `TryParseValueFromString`, and `FormatNullableValue`.

File: src/Core/Components/NumberInput/FluentNumberInput.razor.cs (hunk 13 above)
Comment (AClerbois) [reply src/Core/Components/NumberInput/FluentNumberInput.razor.cs]: Addressed differently: replaced the `IComparable<TValue>` constraint with `INumber<TValue>` (from `System.Numerics`, available since .NET 7). Since `INumber<T>` inherits `IFormattable`, we now call `value.ToString(null, Culture)` directly — no boxing occurs because `TValue` is constrained to `struct`. This avoids the large `switch` on concrete types while covering all numeric types.

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor (hunk 19 above)
Comment (AClerbois) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor:10]: This is a limitation of the `<fluent-text-input type="number">` web component — it doesn't properly support decimal character input like a native `<input type="number">` does. As @MarvinKlein1508 mentioned, switching to `type="text"` may be the only workaround. This would require changes to the Razor template and JS interop. Should we address this in a follow-up PR?

File: src/Core/Components/NumberInput/FluentNumberInput.razor.cs (hunk 12 above)
Comment (AClerbois) [reply src/Core/Components/NumberInput/FluentNumberInput.razor.cs]: Done. Added `NumberInput_InvalidValue` key to `LanguageResource.resx` with value `"The {0} field is not a valid number."` and the error message now uses `Localizer[Localization.LanguageResource.NumberInput_InvalidValue]`.

File: examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor (hunk 19 above)
Comment (MarvinKlein1508) [reply examples/Demo/FluentUI.Demo.Client/Documentation/Components/NumberInput/Examples/NumberInputCulture.razor:10]: @AClerbois I'm not sure about a follow-up PR. Right now this component is pretty much unusable for me when I need to enter and decimal numbers. It only works for full numbers for me.

I can't enter `1.5` or `1,5`. My locale is de-de

**Comment by AClerbois** [comment]
I’m working on it
